engine/src/main/java/org/terasology/recording/RecordAndReplaySerializer.java

```diff
@@ -0,0 +1,150 @@
+    private static final String FILE_AMOUNT = "/file_amount" + JSON;
+
+    private RecordAndReplaySerializer() {
+
+    }
+
+    public static void setEntityManager(EntityManager manager) {
+        entityManager = manager;
+    }
+
+    /**
+     * Serialize RecordedEvents, EntityRefIdMap and some RecordAndReplayUtils data.
+     */
+    public static void serializeRecordAndReplayData() {
+        String recordingPath = PathManager.getInstance().getRecordingPath(RecordAndReplayUtils.getGameTitle()).toString();
+        serializeRecordedEvents(recordingPath);
+        Gson gson = new GsonBuilder().create();
+        serializeRefIdMap(gson, recordingPath);
+        serializeFileAmount(gson, recordingPath);
+    }
+
+    /**
+     * Serialize RecordedEvents.
+     * @param recordingPath path where the data should be saved.
+     */
+    public static void serializeRecordedEvents(String recordingPath) {
+        RecordedEventSerializer serializer = new RecordedEventSerializer(entityManager);
+        String filepath = recordingPath + EVENT_DIR + RecordAndReplayUtils.getFileCount() + JSON;
+        RecordAndReplayUtils.setFileAmount(RecordAndReplayUtils.getFileAmount() + 1);
+        RecordAndReplayUtils.setFileCount(RecordAndReplayUtils.getFileCount() + 1);
+        serializer.serializeRecordedEvents(RecordedEventStore.popEvents(), filepath);
+        logger.info("RecordedEvents Serialization completed!");
+    }
+
+    /**
+     * Deserialize RecordedEvents, EntityRefIdMap and some RecordAndReplayUtils data.
+     */
+    public static void deserializeRecordAndReplayData() {
+        String recordingPath = PathManager.getInstance().getRecordingPath(RecordAndReplayUtils.getGameTitle()).toString();
+        deserializeRecordedEvents(recordingPath);
+        Gson gson = new GsonBuilder().create();
+        deserializeRefIdMap(gson, recordingPath);
+        deserializeFileAmount(gson, recordingPath);
+    }
+
+    /**
+     * Deserialize RecordedEvents.
+     * @param recordingPath path where the data was saved.
+     */
+    static void deserializeRecordedEvents(String recordingPath) {
+        RecordedEventSerializer serializer = new RecordedEventSerializer(entityManager);
+        String filepath = recordingPath + EVENT_DIR + RecordAndReplayUtils.getFileCount() + JSON;
+        RecordAndReplayUtils.setFileCount(RecordAndReplayUtils.getFileCount() + 1);
+        RecordedEventStore.setEvents(serializer.deserializeRecordedEvents(filepath));
+        logger.info("RecordedEvents Deserialization completed!");
+    }
+
+    private static void serializeRefIdMap(Gson gson, String recordingPath) {
+        try {
+            JsonWriter writer = new JsonWriter(new FileWriter(recordingPath + REF_ID_MAP));
+            gson.toJson(EntityRefIdMap.getCurrentMap(), HashMap.class, writer);
+            writer.close();
+            logger.info("RefIdMap Serialization completed!");
+        } catch (Exception e) {
+            logger.error(e.getMessage());
```

Comment:
.error("error while serializing",e); This will log the entire stacktrace instead of the message only.